core/tests/unit/ldml/test_transforms.cpp

```diff
@@ -138,6 +146,226 @@ int test_transforms() {
+      assert_equal(l.match_end(U"a\u1A75"), 2);                             // Match
+      assert_equal(l.match_end(U"SomethingSomethingSomethinga\u1A76"), 2);  // Sub-Match
+
+      // generate sort keys
+      std::cout << __FILE__ << ":" << __LINE__ << " - get_sort_key test " << std::endl;
+      std::u32string str = U"a\u1A78";
+      // get the baseline sortkey
+      auto keylist = reorder_sort_key::from(str);
+      std::cout << __FILE__ << ":" << __LINE__ << "  baseline sortkey" << std::endl;
+      for (auto i = keylist.begin(); i < keylist.end(); i++) {
+        i->dump();
+      }
+      // make sure it matches
+      assert_equal(l.match_end(str), 2);
+      // update the keylist with these elements.
+      l.update_sort_key(0, keylist);
+      std::cout << __FILE__ << ":" << __LINE__ << "  updated sortkey" << std::endl;
+      assert_equal(keylist.size(), 2);
+      size_t secondary = 0;
+      for (auto i = keylist.begin(); i < keylist.end(); i++) {
+        i->dump();
+        assert_equal(i->secondary, secondary);
+        assert_equal(i->quaternary, secondary);
+        secondary++;
+      }
+      std::cout << std::endl;
+
+      // spot check first sortkey
+      assert_equal(keylist.begin()->primary, 80);
+      assert_equal(keylist.begin()->tertiary, -12);
+      assert_equal(keylist.begin()->ch, 0x61);
+      std::cout << __FILE__ << ":" << __LINE__ << "  sorted sortkey" << std::endl;
+
+      // now sort them
+      std::sort(keylist.begin(), keylist.end());
+      for (auto i = keylist.begin(); i < keylist.end(); i++) {
+        i->dump();
+      }
+      std::cout << std::endl;
+      // spot check first sort key
+      assert_equal(keylist.begin()->primary, -12);
+      assert_equal(keylist.begin()->tertiary, 55);
+      assert_equal(keylist.begin()->ch, 0x1A78);
+    }
+
+    std::cout << __FILE__ << ":" << __LINE__ << " - key test " << std::endl;
+    {
+      // can we sort an array of reorder sort keys?
+      reorder_sort_key keys0[] = {
+          {U'\u1A21', 0, 0, 0, 0},
+          {U'\u1A6B', 42, 1, 0, 1},
+          {U'\u1A76', 55, 2, 0, 2},
+          {U'\u1A60', 10, 3, 0, 3},
+          {U'\u1A45', 10, 4, 0, 3}};
+      // add it to a list
+      std::deque<reorder_sort_key> keylist;
+      for (size_t i = 0; i < sizeof(keys0) / sizeof(keys0[0]); i++) {
+        keylist.emplace_back(keys0[i]);
+        // print it out
+        keys0[i].dump();
+      }
+      // now sort it
+      std::cout << __FILE__ << ":" << __LINE__ << "  sort" << std::endl;
+      std::sort(keylist.begin(), keylist.end());
+      std::u32string sorted;
+      for (auto i = keylist.begin(); i < keylist.end(); i++) {
+        i->dump();
+        sorted.append(1, i->ch);
+      }
+      // did the roast come out?
+      zassert_string_equal(sorted, expect);
+    }
+    std::cout << "now prepare the reorder elements" << std::endl;
+    transforms tr;
+    {
+      reorder_group rg;
```

Comment:
the following code 'manually' sets up what the loader will be able to load out of KMX+